apps/shared/avifpng.c

```diff
@@ -128,8 +128,13 @@ avifBool avifPNGRead(const char * inputFilename, avifImage * avif, avifPixelForm
     avif->yuvFormat = requestedFormat;
     if (avif->yuvFormat == AVIF_PIXEL_FORMAT_NONE) {
         // Identity is only valid with YUV444.
-        avif->yuvFormat = (avif->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_IDENTITY) ? AVIF_PIXEL_FORMAT_YUV444
-                                                                                          : AVIF_APP_DEFAULT_PIXEL_FORMAT;
+        if (avif->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_IDENTITY) {
+            avif->yuvFormat = AVIF_PIXEL_FORMAT_YUV444;
+        } else if (rawColorType == PNG_COLOR_TYPE_GRAY) {
```

Comment:
Should we also allow `PNG_COLOR_TYPE_GRAY_ALPHA` here?

```
        } else if ((rawColorType == PNG_COLOR_TYPE_GRAY) || (rawColorType == PNG_COLOR_TYPE_GRAY_ALPHA)) {
```


See line 110.

Reply:
Yes, Done.